modules/space/kepler.cpp

```diff
@@ -0,0 +1,637 @@
+        p.period = std::chrono::seconds(std::chrono::hours(24)).count() / meanMotion;
+
+        result.push_back(p);
+
+        lineNum = lineNum + 3;
+    }
+
+    return result;
+}
+
+std::vector<SatelliteParameters> readOmmFile(std::filesystem::path file) {
+    ghoul_assert(std::filesystem::is_regular_file(file), "File must exist");
+
+    std::vector<SatelliteParameters> result;
+
+    std::ifstream f;
+    f.open(file);
+
+    int lineNum = 1;
+    std::optional<SatelliteParameters> current = std::nullopt;
+    std::string line;
+    while (std::getline(f, line)) {
+        if (line.empty()) {
+            continue;
+        }
+
+        // Tokenize the line
+        std::vector<std::string> parts = ghoul::tokenizeString(line, '=');
+        for (std::string& p : parts) {
+            ghoul::trimWhitespace(p);
+        }
+
+        if (parts.size() != 2) {
+            throw ghoul::RuntimeError(fmt::format(
+                "Malformed line '{}' at {}", line, lineNum
+            ));
+        }
+
+        if (parts[0] == "CCSDS_OMM_VERS") {
+            if (parts[1] != "2.0") {
+                LWARNINGC(
+                    "OMM",
+                    fmt::format(
+                        "Only version 2.0 is currently supported but found {}. "
+                        "Parsing might fail",
+                        parts[1]
+                    )
+                );
+            }
+
+            // We start a new value so we need to store the last one...
+            if (current.has_value()) {
+                result.push_back(*current);
+            }
+
+            // ... and start a new one
+            current = SatelliteParameters();
+        }
+
+        ghoul_assert(current.has_value(), "No current element");
+
+        if (parts[0] == "OBJECT_NAME") {
+            current->name = parts[1];
+        }
+        else if (parts[0] == "OBJECT_ID") {
+            current->id = parts[1];
+        }
+        else if (parts[0] == "EPOCH") {
+            current->epoch = epochFromOmmString(parts[1]);
+        }
+        else if (parts[0] == "MEAN_MOTION") {
+            float mm = std::stof(parts[1]);
+            current->semiMajorAxis = calculateSemiMajorAxis(mm);
+            current->period = std::chrono::seconds(std::chrono::hours(24)).count() / mm;
+        }
+        else if (parts[0] == "SEMI_MAJOR_AXIS") {
+
+        }
+        else if (parts[0] == "ECCENTRICITY") {
+            current->eccentricity = std::stof(parts[1]);
+        }
+        else if (parts[0] == "INCLINATION") {
+            current->inclination = std::stof(parts[1]);
+        }
+        else if (parts[0] == "RA_OF_ASC_NODE") {
+            current->ascendingNode = std::stof(parts[1]);
+        }
+        else if (parts[0] == "ARG_OF_PERICENTER") {
+            current->argumentOfPeriapsis = std::stof(parts[1]);
+        }
+        else if (parts[0] == "MEAN_ANOMALY") {
+            current->meanAnomaly = std::stof(parts[1]);
+        }
```

Comment:
Do we need an else case to log an error if `parts[0]` doesn't match any of the above?

Reply:
There are a bunch of optional keys that are defined in the OMM format that we are ignoring, so I think it would get too chatty if we print a message every time